processing/topics/roads_bikelanes/bikelanes/BikelaneCategories.lua

```diff
@@ -337,68 +418,72 @@ local sharedBusLaneBikeWithBus = BikelaneCategory.new({
 --
 -- TODO: Remove this category and make it an exit conditino to "needsClarification"
 local sharedLane = BikelaneCategory.new({
+  name = 'explicitSharedLaneButNoSignage',
   desc = '',
+  infrastrucuteExists = true,
   condition = function(tags)
     if tags.cycleway == "shared" then
-      return "explicitSharedLaneButNoSignage"
+      return true
     end
   end
 })
 
 -- This is where we collect bike lanes that do not have sufficient tagging to be categorized well.
 -- They are in OSM, but they need to be improved, which we show in the UI.
 local needsClarification = BikelaneCategory.new({
+  name = 'needsClarification',
   desc = 'Bike infrastructure that we cannot categories properly due to missing or ambiguous tagging.' ..
       ' Check the `todos` property on hints on how to improve the tagging.',
+  infrastrucuteExists = true,
   condition = function(tags)
     if tags.highway == "cycleway"
         or (tags.highway == "path" and tags.bicycle == "designated") then
-      return "needsClarification"
+      return true
     end
   end
 })
 
-local function defineCategory(tags, categoryDefinitions)
-  for _, categoryDefinition in pairs(categoryDefinitions) do
-    local category = categoryDefinition:checkCondition(tags)
-    if category ~= nil then
+-- The order specifies the precedence; first one with a result win.
+local categoryDefinitions = {
+  dataNo,
+  isSeparate,
+  implicitOneWay,
+  cyclewayLink,
+  crossing,
+  livingStreet,
+  bicycleRoad_vehicleDestination,
+  bicycleRoad,
```

Comment:
```suggestion
  bicycleRoad, -- has to come after `bicycleRoad_vehicleDestination`
```

Da, wo wir wissen, dass die Reihenfolge entscheidend ist, sollten wir es IMO immer dokumentieren.